### PR TITLE
Use random name for nickname by default

### LINKED ISSUE #81: Generate random nick
Generate a random nick when starting the web app so that users can differentiate themselves from other without having to use the `/nick` command

### PR DESCRIPTION
**Pull request by littlealex003**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Feature

- **What is the current behavior?** (You can also link to an open issue here)
- Resolves #81
- Default is "web-chat"

- **What is the new behavior (if this is a feature change)?**
- Random name generated if name is "web-chat"

### CODE REVIEW

File: web-chat/src/App.tsx [hunk 1]
```diff
@@ -9,13 +9,17 @@ import handleCommand from './command';
...
 
 export const ChatContentTopic = 'dingpu';
 
 export default function App() {
   let [stateMessages, setMessages] = useState<ChatMessage[]>([]);
   let [stateWaku, setWaku] = useState<Waku | undefined>(undefined);
   let [nick, setNick] = useState<string>('web-chat');
+  if(nick==='web-chat') {
+    setNick(generate());
+  }
```
Comment (D4nte) [inline web-chat/src/App.tsx:22]:
```suggestion
  let [nick, setNick] = useState<string>(generate());
```

Does this work?

Also, you need to run the linter: `cd web-chat; npm run fix` (the CI should pick on it).

Comment (littlealex003) [reply web-chat/src/App.tsx:22]: seems to work well, yes. i wasn't sure if that string had any side effect other than being the default, but you are right that's better.

**Review by D4nte — COMMENTED** [review]
Looks good, can you please squash your commits in one commit please?

**Comment by littlealex003** [comment]
Sure, sorry I thought Github allowed you to do that automatically when you
accepted a PR (maybe that was Gitlab, I am often using different ones)

**Comment by D4nte** [comment]
> Sure, sorry I thought Github allowed you to do that automatically when you accepted a PR (maybe that was Gitlab, I am often using different ones)

Yes but this does not prevent the need of having a clean commit tree. Squash merge still condensed the commit descriptions of 4 commits into one commit. In the case of this change, a single patch is more appropriate.

**Comment by littlealex003** [comment]
Fair enough. I think I will have to do a force push on the branch to
overwrite it? I will see. The commits are non-sequential; maybe I should
stop using git on the command line and find a tool to help with these
things, it may be easier just to redo it LOL

**Comment by D4nte** [comment]
> Fair enough. I think I will have to do a force push on the branch to overwrite it? I will see. The commits are non-sequential; maybe I should stop using git on the command line and find a tool to help with these things, it may be easier just to redo it LOL


If you do 

```
git rebase --interactive HEAD~5
```

You can enter the interactive mode. From there, you can select `fixup` (`f)` for all commits apart the top one. YOu can also reword the description of the top one.

**Comment by littlealex003** [comment]
Should be good now, take a look.

**Review by D4nte — APPROVED** [review]

**Comment by D4nte** [comment]
bors merge

**Comment by D4nte** [comment]
Please note that `"server-name-generator"` goes under `"dependencies"`, not `devDependencies` as it is use by the actual app.

Only packages uses for tooling and development go under `devDependencies`.

**Comment by littlealex003** [comment]
> Please note that `"server-name-generator"` goes under `"dependencies"`, not `devDependencies` as it is use by the actual app.
> 
> Only packages uses for tooling and development go under `devDependencies`.

Sorry you are correct of course; I must have used the wrong 'npm install' flag. I'm still adjusting to moving from Maven (where I did everything manually) to npm (where it does it for me, but obviously I should check better).

**Comment by D4nte** [comment]
> > Please note that `"server-name-generator"` goes under `"dependencies"`, not `devDependencies` as it is use by the actual app.
> > Only packages uses for tooling and development go under `devDependencies`.
> 
> Sorry you are correct of course; I must have used the wrong 'npm install' flag. I'm still adjusting to moving from Maven (where I did everything manually) to npm (where it does it for me, but obviously I should check better).

No worries, please note I already correct. No need to touch this branch anymore.

**Comment by D4nte** [comment]
Don't worry about the CI failure, I'll sort it out.

**Comment by littlealex003** [comment]
> Don't worry about the CI failure, I'll sort it out.

OK great. I hope I have been useful! :)

**Comment by D4nte** [comment]
bors merge

**Pull request merged**